[cmake] Update supported cmake version

# Summary
Version of supported cmake shall be updated.

# Observed behavior
When building with cmake 3.22.x build fails with error:
`CMake Error at src/CMakeLists.txt:47 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "DEPRECATION" is not allowed.`
Updating cmake to 3.25.x solves the issue.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -44,7 +44,9 @@ function(add_deprecated_library target_name)
   # The INTERFACE IMPORTED target works like an alias which can have different properties
   add_library(deprecated_name INTERFACE IMPORTED)
   target_link_libraries(deprecated_name INTERFACE target_name)
-  set_target_properties(deprecated_name PROPERTIES DEPRECATION "${deprecated_name} target is deprecated, please use ${target_name} instead")
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.22.1")
+    set_target_properties(deprecated_name PROPERTIES DEPRECATION "${deprecated_name} target is deprecated, please use ${target_name} instead")
+  endif()
 endfunction()
 
 

